src/docfx/lib/PathUtility.cs

```diff
@@ -128,6 +131,49 @@ public static void CreateDirectoryFromFilePath(string filePath)
             Directory.CreateDirectory(directoryPath);
         }
 
+        /// <summary>
+        /// Converts an URL to a human readable short directory name
+        /// </summary>
+        public static string UrlToFolderName(string url)
+        {
+            var hash = url.GetMd5HashShort();
+
+            // Trim https://
+            var index = url.IndexOf(':');
+            if (index > 0)
+            {
+                url = url.Substring(index);
+            }
+
+            url = url.TrimStart('/', '\\', '.', ':').Trim();
+
+            var result = new StringBuilder();
+
+            // Take the first 5 segments and the surrounding 8 chars in each segment, then remove invalid path chars.
+            foreach (var segment in url.Split(new[] { '/', '\\', ' ', '?', '#' }, StringSplitOptions.RemoveEmptyEntries).Take(5))
```

Comment:
What about take the last 5 segments? When downloading from URL, the last part can tell what the file actually is.

Reply:
Maybe also use surrounding segments